haskell-src/Concordium/Wasm.hs

```diff
@@ -229,6 +335,38 @@ isValidReceiveName proposal =
       hasDot = Text.any (== '.') proposal
   in hasValidLength && hasValidCharacters && hasDot
 
+-- |Name of the entrypoint, i.e., the part of the receive name after the dot.
+newtype EntrypointName = EntrypointName { entrypointName :: Text }
+    deriving (Eq, Show, Ord)
+    deriving(AE.ToJSON) via Text
+
+-- |Check whether the given text is a valid entrypoint name.
+-- This is the case if
+--
+-- * length is <= maxFuncNameSize
```

Comment:
We use the same limit for `ReceiveName`, which also has the `<contract_name>.` prefix. So I'm not sure this is correct.

Reply:
Well it is correct in as much as it is a definition of what a valid entrypoint name is. You could argue whether it is a sensible definition. We could say `< maxFuncNameSize` to account for the `.` but that does not really improve anything.